src/site/stages/build/drupal/graphql/eventPage.graphql.js

```diff
@@ -77,3 +78,51 @@ module.exports = `
     fieldAdditionalInformationAbo {processed}
  }
 `;
+
+const firstDayOfMonthUnixStamp = moment()
+  .second(0)
+  .minute(0)
+  .hour(0)
+  .date(1)
+  .unix();
+
+const GetNodeEventPages = `
+  ${eventPage}
+
+  query GetNodeEventPages($onlyPublishedContent: Boolean!) {
+    nodeQuery(limit: 1000, filter: {
+      conditions: [
+        { field: "status", value: ["1"], enabled: $onlyPublishedContent },
+        { field: "field_datetime_range_timezone.end_value", value: ["${firstDayOfMonthUnixStamp}"], operator: GREATER_THAN },
+      ]
+    }) {
+      entities {
+        ... eventPage
+      }
+    }
+  }
+`;
+
+const GetArchivedNodeEventPages = `
+${eventPage}
+
+query GetNodeEventPages($onlyPublishedContent: Boolean!) {
+  nodeQuery(limit: 1000, filter: {
+    conditions: [
+      { field: "status", value: ["1"], enabled: $onlyPublishedContent },
+      { field: "type", value: ["event"] },
+      { field: "field_datetime_range_timezone.end_value", value: ["${firstDayOfMonthUnixStamp}"], operator: SMALLER_THAN_OR_EQUAL },
+    ]
+  }) {
+    entities {
+      ... eventPage
+    }
+  }
+}
```

Comment:
`GetNodeEventPages` was expensive, so I decided to try making it more efficient by splitting it into two. The second query gets all events that ended before the start of this month, with the hypothesis being that that event data is unlikely to change and therefore a good candidate to be cached. Then `GetNodeEventPages` just gets whatever events are left over. Not certain that it worked yet but seems promising so far.